### PR TITLE
Add codecov integration

### PR DESCRIPTION
**Pull request by ssbarnea**

### CODE REVIEW

File: playbooks/post-run.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+- hosts: all:!appliance
+  tasks:
+    - name: Coverage report
+      environment:
+        CODECOV_TOKEN: "{{ codecov.token }}"
+      command:
+        set -ex
+        pip install 'coverage[toml]'
+        coverage combine .tox/.coverage.*
+        curl -Os https://uploader.codecov.io/latest/linux/codecov
```
Comment (webknjaz) [inline playbooks/post-run.yaml:28]: There's a curated executable that ansible-core uses. You should probably use it too.

File: playbooks/post-run.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          UT5ImZ8kwOGFnpvJyJsJy747dJfLiqLSxIgTjL3b4hpHJ/vHhvgaZyGrKeNHHs=
+
+- hosts: all:!appliance
+  tasks:
+    - name: Coverage report
+      environment:
+        CODECOV_TOKEN: "{{ codecov.token }}"
+      command:
+        set -ex
+        pip install 'coverage[toml]'
```
Comment (webknjaz) [inline playbooks/post-run.yaml:26]: No need for toml. Coveragepy should use its own config as it's a de-facto practice used across other projects.

File: playbooks/post-run.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        coverage combine .tox/.coverage.*
```
Comment (webknjaz) [inline playbooks/post-run.yaml:27]: This shouldn't be necessary unless tox itself is executed in parallel mode. AFAIK only tox is.

File: tox.ini [hunk 4]
```diff
@@ -30,6 +30,7 @@ commands =
   /bin/bash -c 'py.test -vvvv -n `python -c "import multiprocessing; import os; print(int(os.environ.get(\"XDIST_CPU\") or multiprocessing.cpu_count()))"` --dist=loadfile --maxfail=15 --durations=100 --cov ansible_navigator --cov share --cov-report term-missing --cov-branch --showlocals'
 setenv =
   TERM = xterm-256color
+  COVERAGE_FILE = {env:COVERAGE_FILE:{toxworkdir}/.coverage.{envname}}
```
Comment (webknjaz) [inline tox.ini:33]: I don't think this is necessary. I suppose you may be looking for an XML report. In which case, add `--cov-report xml` maybe?

Comment (ssbarnea) [reply tox.ini:33]: Not needed for codecov itselt but it helps decluttering project root, so we don't add extra temp files to it.

Comment (ssbarnea) [reply tox.ini]: Removed it as is not directly needed by codecov.

File: .github/workflows/tox.yml [hunk 5]
```diff
@@ -66,6 +66,14 @@ jobs:
           python3 -m tox
         env:
           TOXENV: ${{ matrix.tox_env }}
+      - name: Upload coverage data
+        uses: codecov/codecov-action@v2
+        with:
+          directory: .tox
+          files: 'coverage-*.xml'
+          name: ${{ matrix.tox_env }}
```
Comment (webknjaz) [inline .github/workflows/tox.yml:74]: I also recommend adding flags — they are useful for determining which jobs/envs covered specific lines.

File: tox.ini (hunk 4 above)
Comment (webknjaz) [reply tox.ini]: It is, unless you configure it to use a different report format IIRC

**Review by webknjaz — CHANGES_REQUESTED** [review]
Urgh... Yet another unapproved merge 🤦🏼

**Pull request merged**